### PR TITLE
Fabo/vue cli 3

### PR DESCRIPTION
**Pull request by faboweb**
Closes #ISSUE

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -3,12 +3,7 @@ module.exports = {
   env: {
     node: true
   },
-  extends: [
-    "plugin:vue/recommended",
-    "eslint:recommended",
-    "prettier/vue",
-    "plugin:prettier/recommended"
-  ],
+  extends: ["plugin:vue/recommended", "@vue/prettier"],
```
Comment (jbibla) [inline .eslintrc.js]: 😅

File: .gitignore [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
-tasks/builds
-coverage
 node_modules
-npm-debug.log
-npm-debug.log.*
-/package-lock.json
-thumbs.db
-!.gitkeep
-.vscode/launch.json
-yarn-error.log
```
Comment (jbibla) [inline .gitignore]: are you sure these can all be deleted?

File: package.json [hunk 3]
```diff
@@ -1,120 +1,32 @@
 {
   "name": "lunie",
-  "productName": "lunie",
   "version": "1.0.0-beta.104",
   "description": "Lunie is the staking and governance platform for proof-of-stake blockchains.",
   "author": "Lunie International Software Systems Inc. <hello@lunie.io>",
-  "license": "Apache-2.0",
```
Comment (jbibla) [inline package.json]: why remove this?

File: package.json [hunk 4]
```diff
@@ -125,44 +37,79 @@
...
     "js-beautify": "1.9.1",
+    "lodash.isempty": "^4.4.0",
+    "lodash.orderby": "^4.6.0",
+    "lodash.uniqby": "^4.7.0",
     "moment": "2.24.0",
     "ndjson": "1.5.0",
     "nib": "1.1.2",
     "no-scroll": "2.1.1",
     "old": "0.2.0",
-    "perfect-scrollbar": "1.4.0",
```
Comment (jbibla) [inline package.json]: 🎉

File: package.json [hunk 5]
```diff
@@ -125,44 +37,79 @@
     "bignumber.js": "8.1.1",
     "camelcase": "5.3.0",
     "copy-webpack-plugin": "5.0.4",
+    "core-js": "^2.6.5",
     "crypto-js": "3.1.9-1",
     "js-beautify": "1.9.1",
+    "lodash.isempty": "^4.4.0",
+    "lodash.orderby": "^4.6.0",
+    "lodash.uniqby": "^4.7.0",
     "moment": "2.24.0",
```
Comment (jbibla) [inline package.json]: would be great to import moment functions instead of the whole library - it's huge.

File: package.json [hunk 6]
```diff
@@ -125,44 +37,79 @@
     "bignumber.js": "8.1.1",
     "camelcase": "5.3.0",
     "copy-webpack-plugin": "5.0.4",
```
Comment (jbibla) [inline package.json]: shouldn't be a dependency (maybe devDep?)

File: package.json [hunk 7]
```diff
@@ -1,120 +1,32 @@
...
+    "test:unit": "vue-cli-service test:unit",
+    "testnet:start": "npm run testnet:start:container && npm run proxy",
+    "testnet:start:container": "docker run --rm -p 26657:26657 --name testnet -p 9070:9070 --detach -e MAX_NODES=${MAX_NODES:-2} lunieio/testnet:v0.34.7",
+    "testnet:stop": "docker stop testnet"
   },
   "dependencies": {
+    "@babel/polyfill": "^7.4.4",
     "@lunie/cosmos-api": "0.1.2",
     "@lunie/cosmos-keys": "0.1.0",
     "@lunie/cosmos-ledger": "0.0.5",
```
Comment (jbibla) [inline package.json]: line below this is sentry related - can be removed

File: package.json [hunk 8]
```diff
@@ -125,44 +37,79 @@
...
+    "cross-env": "^5.2.0",
+    "csp-webpack-plugin": "^2.0.2",
+    "eslint": "^5.16.0",
+    "eslint-plugin-prettier": "^3.1.0",
+    "eslint-plugin-vue": "^5.0.0",
+    "jest-localstorage-mock": "^2.4.0",
+    "kill-port": "^1.5.1",
+    "live-server": "^1.2.1",
+    "live-server-https": "^0.0.2",
+    "lodash.trim": "^4.5.1",
```
Comment (jbibla) [inline package.json]: devDep?

File: src/initializeApp.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
+/* istanbul ignore file: really just integrations */
+
+import { listenToExtensionMessages } from "scripts/extension-utils"
+import {
+  enableGoogleAnalytics,
+  setGoogleAnalyticsPage
+} from "scripts/google-analytics"
+import * as Sentry from "@sentry/browser"
```
Comment (jbibla) [inline src/initializeApp.js]: lots of sentry stuff still here. should we leave it?

File: src/main.js [hunk 10]
```diff
@@ -1,9 +1,30 @@
...
+import App from "./App.vue"
+import init from "./initializeApp"
+import { getURLParams } from "scripts/url.js"
+import "./registerServiceWorker"
+import "@babel/polyfill"
+
+Vue.config.productionTip = false
+
+Vue.use(Tooltip, { delay: 1 })
+Vue.use(Vuelidate)
```
Comment (jbibla) [inline src/main.js]: being used in this PR still?

File: package.json (hunk 5 above)
Comment (colw) [reply package.json]: We could use date-fns? https://date-fns.org/

It's a more functional design that doesn't mutate state, and allegedly smaller.

File: src/vuex/modules/delegates.js [hunk 11]
```diff
@@ -1,6 +1,6 @@
 import * as Sentry from "@sentry/browser"
 import BN from "bignumber.js"
-import b32 from "scripts/b32"
+import b32 from "scripts/b32.js"
```
Comment (colw) [inline src/vuex/modules/delegates.js]: Is the `.js` necessary?

File: src/vuex/modules/delegates.js [hunk 12]
```diff
@@ -104,7 +101,6 @@ export default ({ node }) => {
         commit(`setDelegates`, validators)
         commit(`setDelegateLoading`, false)
         dispatch(`updateSigningInfo`, validators)
-        dispatch(`getRewardsFromMyValidators`)
```
Comment (jbibla) [inline src/vuex/modules/delegates.js]: not needed?

File: tests/unit/helpers/emptyModule.js [hunk 13]
```diff
@@ -0,0 +1,6 @@
+// use to map any file or extension to an empty module
+module.exports = {
+  process() {
+    return ""
```
Comment (jbibla) [inline tests/unit/helpers/emptyModule.js]: what is this for? when is `emptyModule` used?

File: tests/unit/specs/components/common/__snapshots__/TmSessionHardware.spec.js.snap [hunk 14]
```diff
@@ -1,27 +1,80 @@
...
-    Use my Ledger Nano
-  </h2> <div class=\\"session-main\\"><div class=\\"tm-hardware-state\\"><div class=\\"tm-hardware-state-content\\"><!----> <div class=\\"tm-hardware-state__label\\"><p>
-            Please plug in your Ledger&nbsp;Nano and open the Cosmos Ledger
-            app
-          </p> <!----> <button class=\\"tm-btn\\"><span class=\\"tm-btn__container\\"><!----><span class=\\"tm-btn__value\\">Sign In</span></span></button></div> <img src=\\"~assets/images/loader.svg\\" alt=\\"a small spinning circle to display loading\\" style=\\"display: none;\\"></div></div></div></div></div> <button class=\\"tm-btn session-close\\"><span class=\\"tm-btn__container tm-btn--secondary\\"><!----><span class=\\"tm-btn__value\\">Back to Lunie</span></span></button></div>"
+<div class="session-frame" name="component-fade" mode="out-in">
```
Comment (jbibla) [inline tests/unit/specs/components/common/__snapshots__/TmSessionHardware.spec.js.snap]: woah this is awesome

File: tests/unit/specs/components/wallet/PageWallet.spec.js [hunk 15]
```diff
@@ -1,5 +1,5 @@
 import { shallowMount, createLocalVue } from "@vue/test-utils"
-import mockValues from "test/unit/helpers/mockValues.js"
+import mockValues from "tests/unit/helpers/mockValues.js"
```
Comment (jbibla) [inline tests/unit/specs/components/wallet/PageWallet.spec.js]: we're still using `mockValues`? 👃

File: tests/unit/specs/utils/utils.spec.js [hunk 16]
```diff
@@ -1,4 +1,4 @@
-import { roundObjectPercentages } from "utils"
+import { roundObjectPercentages } from "src/utils"
```
Comment (jbibla) [inline tests/unit/specs/utils/utils.spec.js]: i wish this wasn't called `utils` 😝

File: vue.config.js [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+        new webpack.DefinePlugin({
+          "process.env": {
+            NODE_ENV: JSON.stringify(process.env.NODE_ENV),
+            RPC: JSON.stringify(process.env.RPC),
+            STARGATE: JSON.stringify(process.env.STARGATE),
+            RELEASE: JSON.stringify(commitHash),
+            GOOGLE_ANALYTICS_UID: JSON.stringify(
+              process.env.GOOGLE_ANALYTICS_UID
+            ),
+            SENTRY_DSN: JSON.stringify(process.env.SENTRY_DSN)
```
Comment (jbibla) [inline vue.config.js]: can be removed?

File: vue.config.js [hunk 18]
```diff
@@ -0,0 +1,108 @@
...
+          "default-src": `'self'`,
+          "script-src": [
+            `'self'`,
+            `https://app.appzi.io/`,
+            `https://*.lunie.io`
+          ],
+          "worker-src": `'none'`,
+          // 'style-src': production ? `'self'` : `*`, // SECURITY Appzi is applying styles inline, inquired to them already
+          "style-src": [`'self'`, `'unsafe-inline'`],
+          "connect-src": [
```
Comment (jbibla) [inline vue.config.js]: how careful should we be here? 

can we only leave links here that we absolutely need?

File: vue.config.js [hunk 19]
```diff
@@ -0,0 +1,108 @@
...
+            ...[process.env.STARGATE].filter(x => x !== undefined),
+            ...[process.env.RPC]
+              .filter(x => x !== undefined)
+              .map(x => x.replace("https", "wss"))
+          ],
+          "frame-src": [`'self'`, `https://app.appzi.io/`],
+          "img-src": [
+            `'self'`,
+            `https://www.google-analytics.com/`,
+            `https://s3.amazonaws.com/keybase_processed_uploads/`
```
Comment (jbibla) [inline vue.config.js]: still needed?

File: package.json (hunk 5 above)
Comment (jbibla) [reply package.json]: whatever is easiest and accomplishes the goal!

File: src/main.js (hunk 10 above)
Comment (colw) [reply src/main.js]: We still use Vuelidate

File: tests/unit/specs/utils/utils.spec.js (hunk 16 above)
Comment (colw) [reply tests/unit/specs/utils/utils.spec.js]: How about `misc`? 😄 

It's only the one function and we could move it to `num.js`

File: src/main.js (hunk 10 above)
Comment (jbibla) [reply src/main.js]: ah sorry. vuetify was on my mind.

File: tests/unit/specs/utils/utils.spec.js (hunk 16 above)
Comment (jbibla) [reply tests/unit/specs/utils/utils.spec.js]: if it's only one function, yeah let's move it!  (can be done later - i'm just being nit picky)

File: package.json (hunk 5 above)
Comment (colw) [reply package.json]: Not for this PR. Happy to convert after.

File: .gitignore (hunk 2 above)
Comment (faboweb) [reply .gitignore]: Some of them are still in. Let's add again when we see fit to avoid polution

File: tests/unit/helpers/emptyModule.js (hunk 13 above)
Comment (faboweb) [reply tests/unit/helpers/emptyModule.js]: i.e. for `.css` files which we don't want to import in jest

File: vue.config.js (hunk 17 above)
Comment (faboweb) [reply vue.config.js]: removing sentry in another PR. would mean a lot more work.

File: vue.config.js (hunk 18 above)
Comment (faboweb) [reply vue.config.js]: removed testnet

File: vue.config.js (hunk 19 above)
Comment (faboweb) [reply vue.config.js]: that's the actual source of the images uploaded by keybase

**Pull request merged**